### PR TITLE
allocate some dummy variables to pass debugging mode

### PR DESCRIPTION
**Pull request by weiyuan-jiang**
In debugging mode,  the subroutines need allocated arguments.

### CODE REVIEW

**Comment by weiyuan-jiang** [comment]
This PR addresses  #414 too

File: src/Components/GEOSldas_GridComp/Shared/my_matrix_functions.F90 [hunk 1]
```diff
@@ -15,6 +15,7 @@ module my_matrix_functions
   !public :: adjust_mean
   !public :: matrix_std
   public :: unique_rows_3col
+  public :: unique_rows_2col
```
Comment (gmao-rreichle) [inline src/Components/GEOSldas_GridComp/Shared/my_matrix_functions.F90:18]: @weiyuan-jiang : After you are done with testing the subroutines, we should comment out this line again or enclose it within a pre-processor if statement (i.e., `#if 0 ... #endif` before we merge the branch.  The subroutine unique_rows_2col() is not used anywhere else in the code (besides the unit test program).

**Comment by gmao-rreichle** [comment]
@weiyuan-jiang: Once we have confirmation that the nightly tests pass, I'm ok with merging this PR.  Thanks!
cc: @biljanaorescanin

File: src/Components/GEOSldas_GridComp/Shared/my_matrix_functions.F90 (hunk 1 above)
Comment (weiyuan-jiang) [reply src/Components/GEOSldas_GridComp/Shared/my_matrix_functions.F90]: > @weiyuan-jiang : After you are done with testing the subroutines, we should comment out this line again or enclose it within a pre-processor if statement (i.e., `#if 0 ... #endif` before we merge the branch. The subroutine unique_rows_2col() is not used anywhere else in the code (besides the unit test program).

I commented it out

**Comment by mathomp4** [comment]
I think everything has been passing at night. I haven't noticed any fails.

**Comment by gmao-rreichle** [comment]
> I think everything has been passing at night. I haven't noticed any fails.

@mathomp4: This PR hasn't been merged yet, so the successful nightly tests don't tell us about the 0-diff nature of the branch associated with the PR.

@weiyuan-jiang: Did you test this branch?  Or should @biljanaorescanin do it?  We need only one of you to test it, but so far we don't have confirmation of a successful test.  Perhaps I missed it?

**Comment by biljanaorescanin** [comment]
@gmao-rreichle  I am testing it right now...submitted test at 10am so I will know in an hour or a bit less.

**Comment by mathomp4** [comment]
@gmao-rreichle Well I feel dumb. Thankfully it's Friday 😄

**Comment by gmao-rreichle** [comment]
Thanks, @biljanaorescanin.  Are you also running the "assim" cases with in "debug" mode?  The failure of the "assim" cases to run in "debug" mode is what prompted the PR.  
I think @weiyuan-jiang was able to run after inserting allocate(xxx(1)) statements, but we agreed to try allocate(xxx(0)) instead.  Before we can merge the PR, we also need to make sure that the branch runs for the "assim" cases in "debug" mode.  I'm not sure if we have a baseline/reference for this case, but that's not a problem.  The important point is to be able to run "assim" in "debug" mode.

**Comment by biljanaorescanin** [comment]
@gmao-rreichle, I understood @weiyuan-jiang has run that, and I need to run just regular nightly tests. @weiyuan-jiang am I right?

**Comment by gmao-rreichle** [comment]
@biljanaorescanin : I haven't heard from Weiyuan that he has successfully run the "assim" "debug" case after changing to allocation of 0-size arrays, nor have I heard that he's run the nightly test.  He probably got side-tracked with other things, and he's on leave this afternoon.  There's no urgency.  Please do **not** merge the PR until we hear back from Weiyuan.

**Comment by biljanaorescanin** [comment]
> @biljanaorescanin : I haven't heard from Weiyuan that he has successfully run the "assim" "debug" case after changing to allocation of 0-size arrays, nor have I heard that he's run the nightly test. He probably got side-tracked with other things, and he's on leave this afternoon. There's no urgency. Please do **not** merge the PR until we hear back from Weiyuan.

Sure

**Comment by biljanaorescanin** [comment]
All tests passed for this PR. @weiyuan-jiang also confirmed that he has successfully run the "assim" "debug" case after changing to allocation of 0-size arrays.

**Review by gmao-rreichle — APPROVED** [review]

**Pull request merged**